.github/workflows/main.yaml

```diff
@@ -0,0 +1,79 @@
+name: Main
+on:
+  pull_request:
+  push:
+    branches:
+      - 'main'
+    tags:
+      - '*'
+jobs:
+  build:
+    name: build
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-go@v2
+        with:
+          go-version: 1.15
+      - run: make test
+      - run: make install GOBIN=/tmp/
+      - name: Upload binaries for setup-hw
+        uses: actions/upload-artifact@v2
+        with:
+          name: setup-hw-binaries
+          path: |
+            /tmp/setup-hw
+            /tmp/monitor-hw
+            /tmp/collector
+          retention-days: 1
+  build_image:
+    name: build_image
+    needs: build
+    runs-on: ubuntu-20.04
+    steps:
+      - uses: actions/checkout@v2
+      - name: Download binaries for setup-hw
+        uses: actions/download-artifact@v2
+        with:
+          name: setup-hw-binaries
+      - run: |
+          cp setup-hw docker
+          cp monitor-hw docker
+          cp collector docker
+      - run: |
+          docker build -t quay.io/cybozu/setup-hw:latest --target stage1 docker
+          docker build -t quay.io/cybozu/setup-hw-secret:latest docker
+      - run: |
+          mkdir -p /tmp/docker
+          docker save quay.io/cybozu/setup-hw:latest -o /tmp/docker/setup-hw-image
+          docker save quay.io/cybozu/setup-hw-secret:latest -o /tmp/docker/setup-hw-secret-image
+      - name: Upload setup-hw images
+        uses: actions/upload-artifact@v2
+        with:
+          name: setup-hw-images
+          path: |
+            /tmp/docker/setup-hw-image
+            /tmp/docker/setup-hw-secret-image
+          retention-days: 1
+  push_image:
+    name: Push images to quay.io
+    runs-on: ubuntu-20.04
+    if: github.event_name != 'pull_request'
```

Comment:
`needs: build_image`?

Reply:
You added additional events on commits to the `main` branch, and this
condition allows the events to pass through.  Do you really want to 
make a release on commits to the `main` branch?

Reply:
@ymmt2005 

Thanks!
As you pointed out, it was also being executed by the CI running on the main branch, 
so I fixed it by using the condition, ` if: startsWith(github.ref, 'refs/tags/v')`

Reply:
As both `github.ref_type` and `github.event.ref_type` were empty, I will use `github.ref` to know if the workflow is triggered by `git tag`